### ISSUE TITLE
3D Window Inconsistently Freeze on MacOS Big Sur and Catalina

### ISSUE THREAD

**Issue by mrtelecaster** [amethyst/amethyst#2673]
## Description

When I run my own 3D code or any of amethyst's examples, half the time it runs perfectly fine, the other half of the time it renders the initial frame and then appears to freeze while the following warning starts printing to my terminal on loop:

```
[WARN][gfx_backend_metal::window] No frame is available
```

The game loop does appear to be running in the background, but it's just unable to update the image in the window for whatever reason. When I resize the window, it renders a new frame when I release the mouse, and then freezes again. I've primarily been messing with this in the `debug_lines` example, but I've observed the same behaviour in the `animation` example as well. I haven't tested this in 2D.

## Reproduction Steps

1. Run any of the 3D examples on OSX with the Metal rendering backend (I'm going to use `debug_lines` as the example in my reproduction description)
2. If the window does not freeze on first try, close window and run again until one of the runs freezes

## What You Expected to Happen

I expect to see the purple "pendulum" swinging, and expect to be able to use WASD and mouse movements to fly the camera around the environment

## What Actually Happened

The pendulum does not swing and I cannot move the camera. When resizing the window, on releasing the mouse button it will render one new frame with all of the changes that have taken place in the intervening time (pendulum moves, and camera moves to where I would expect from the inputs that I gave it while frozen). After drawing this frame, it returns to a frozen state, while filling my terminal with the `No frame is available` warning.

## Your Environment

Operating System: MacOS Big Sur (Also tested this on Catalina and found the same results, with it freezing much more frequently than on Big Sur)
Operating System Version: 11.2.3
Version of Amethyst: 0.15.3

**Comment by mrtelecaster** [comment]
I meant to write in the original issue, I did find issue #1927 which seems to be a similar issue, but I'm unsure if it's the same as they're describing it freezing when switching between fullscreen windows, where I'm seeing it happen roughly 50% of the time whether or not my terminal is fullscreen. It's also stated on that issue that it's *expected* to be fixed with #2322 but that issue has been stale for close to a year now.

There is also #2236 which describes the same problem happening when you switch screens or move another window in front of the amethyst issue, but I'm seeing this happen when the amethyst window is completely unobstructed, and when I'm running the example program from a terminal that's on the same desktop as the amethyst window, so no desktop/screen switching occurs

**Comment by mrtelecaster** [comment]
Update after playing around with it some more:

I left it overnight and restarted my computer this morning, and now I'm no longer seeing the behavior I originally described. It still freezes, but now it freezes in the circumstances that are described in #1927 and #2322 and I'm unable to replicate what I saw yesterday. Now for me the window freezes when I run the code from a fullscreen IDE or terminal in another desktop, as well as when another window moves in front of the amethyst game window on the main desktop. I haven't done anything since I wrote my original github issue besides restart my computer after updating to Big Sur. While I can't reproduce it anymore, I'm still certain I saw different behavior yesterday, as I tested it very thoroughly to make sure it wasn't the same as the issues I mentioned. So this might be a duplicate but hopefully it can add a bit of new information/context to the problem
